exchange/tx.go

```diff
@@ -494,7 +487,30 @@ func (tx *Tx) QueryFrom(info peer.AddrInfo, key string) error {
 }
 
 // Execute starts a retrieval operation for a given offer and returns the deal ID for that operation
-func (tx *Tx) Execute(of deal.Offer) error {
+func (tx *Tx) Execute(of deal.Offer) TxResult {
+	result := make(chan TxResult, 1)
+	tx.unsub = tx.retriever.SubscribeToEvents(func(event client.Event, state deal.ClientState) {
+		switch state.Status {
+		case deal.StatusCompleted:
+			select {
+			case result <- TxResult{
```

Comment:
why do you use a select/case to send the struct?

Reply:
It is possible that the events gets triggered twice for some reason and so instead of blocking the routine, the switch statement just goes to default and the callback returns without sending to the channel